internal/security/security_test.go

```diff
@@ -0,0 +1,145 @@
+package security
+
+import (
+	"fmt"
+	"strings"
+	"testing"
+
+	"github.com/sourcegraph/sourcegraph/internal/conf"
+	"github.com/sourcegraph/sourcegraph/schema"
+	"github.com/stretchr/testify/assert"
+)
+
+// setMockPasswordPolicyConfig helper for returning customized mock config
+func setMockPasswordPolicyConfig(policyEnabled bool, authPolicyEnabled bool, authMinPasswordLength int,
+	authPolicySpChr int, reqNumber bool, reqCase bool) {
+
+	conf.Mock(&conf.Unified{
+		SiteConfiguration: schema.SiteConfiguration{
+			AuthMinPasswordLength: authMinPasswordLength,
+			ExperimentalFeatures: &schema.ExperimentalFeatures{
+				PasswordPolicy: &schema.PasswordPolicy{
+					Enabled:                   policyEnabled,
+					NumberOfSpecialCharacters: 3,
+					// invert reqNumber and reqCase so it differs AuthPasswordPolicy
+					RequireUpperandLowerCase: !reqNumber,
+					RequireAtLeastOneNumber:  !reqCase,
+				},
+			},
+			AuthPasswordPolicy: &schema.AuthPasswordPolicy{
+				Enabled:                   authPolicyEnabled,
+				NumberOfSpecialCharacters: authPolicySpChr,
+				RequireAtLeastOneNumber:   reqNumber,
+				RequireUpperandLowerCase:  reqCase,
+			},
+		},
+	})
+}
+
+func TestGetPasswordPolicy(t *testing.T) {
+
+	authPolicyLength := 12
+	authPolicySpChr := 2
+
+	setMockPasswordPolicyConfig(false, true,
+		authPolicyLength, authPolicySpChr, true, true)
+
+	t.Run("Fetch correct policy.", func(t *testing.T) {
+		p := conf.AuthPasswordPolicy()
+
+		assert.True(t, p.Enabled)
+		assert.Equal(t, p.MinimumLength, authPolicyLength)
+		assert.Equal(t, p.RequireUpperandLowerCase, true)
+
+		// create experimental policy for testing backwards compatability
+		conf.Mock(&conf.Unified{
+			SiteConfiguration: schema.SiteConfiguration{
+				AuthMinPasswordLength: authPolicyLength,
+				ExperimentalFeatures: &schema.ExperimentalFeatures{
+					PasswordPolicy: &schema.PasswordPolicy{
+						Enabled:                   true,
+						NumberOfSpecialCharacters: authPolicySpChr,
+						RequireUpperandLowerCase:  true,
+						RequireAtLeastOneNumber:   true,
+					},
+				},
+			},
+		})
+
+		p = conf.AuthPasswordPolicy()
+
+		assert.True(t, p.Enabled)
+		assert.Equal(t, p.MinimumLength, authPolicyLength)
+		assert.Equal(t, p.RequireUpperandLowerCase, true)
+		assert.Equal(t, p.NumberOfSpecialCharacters, authPolicySpChr)
+
+	})
+}
+
+func TestFetchPasswordPolicy_nil(t *testing.T) {
```

Comment:
I don't think you need a subtest in this case here (i.e. remove the `t.Run` further down). You could rename the test and that's all the description required, I think.

```suggestion
func TestFetchPasswordPolicyReturnsNil(t *testing.T) {
```